src/main/kotlin/org/acejump/session/Session.kt

```diff
@@ -246,6 +249,23 @@ class Session(private val editor: Editor) {
     }
   }
 
+  private fun calculateMarkAndQuery(jumpPerformed: Boolean): Pair<String?, String?> {
+    var mark: String? = null
+    var query: String? = null
+    if (jumpPerformed) {
+      val offset = editor.caretModel.offset
+      mark = tagger.getMarkByOffset(offset)
+      if (mark != null) {
+        (searchProcessor?.query as? SearchQuery.Literal)?.rawText?.let {
+          if (it.endsWith(mark)) {
+            query = it.dropLast(mark.length)
```

Comment:
This is kinda messy. If we are going to notify consumers of the query and tag selected, it would probably be better to notify listeners as soon as the [the jump is triggered](https://github.com/acejump/AceJump/blob/2c0a236ed84df23b5ab353657fe251157b567d9f/src/main/kotlin/org/acejump/search/Tagger.kt#L56) (since we have the `entry`) rather than waiting until the last minute to reconstruct the query and tag from stale data prior to disposal.